src/utils/vcs.rs

```diff
@@ -595,15 +610,19 @@ fn extract_pr_head_sha_from_event(json_content: &str) -> Option<String> {
 }
 
 /// Extracts the PR base SHA from GitHub Actions event payload JSON.
-/// Returns Ok(None) if not a PR event or if SHA cannot be extracted.
-/// Returns an error if we cannot parse the JSON.
-fn extract_pr_base_sha_from_event(json_content: &str) -> Result<Option<String>> {
-    let v: Value = serde_json::from_str(json_content)
-        .context("Failed to parse GitHub event payload as JSON")?;
+/// Returns None if not a PR event or if SHA cannot be extracted.
+fn extract_pr_base_sha_from_event(json_content: &str) -> Option<String> {
+    let v: Value = match serde_json::from_str(json_content) {
+        Ok(v) => v,
+        Err(_) => {
+            debug!("Failed to parse GitHub event payload as JSON");
+            return None;
+        }
+    };
 
-    Ok(v.pointer("/pull_request/base/sha")
+    v.pointer("/pull_request/base/sha")
         .and_then(|s| s.as_str())
-        .map(|s| s.to_owned()))
+        .map(|s| s.to_owned())
 }
```

Comment:
I see no reason for this function to change. If we want to debug-log the error message rather than error, the calling code should change how it handles the error returned from this function.

```suggestion
/// Extracts the PR base SHA from GitHub Actions event payload JSON.
/// Returns Ok(None) if not a PR event or if SHA cannot be extracted.
/// Returns an error if we cannot parse the JSON.
fn extract_pr_base_sha_from_event(json_content: &str) -> Result<Option<String>> {
    let v: Value = serde_json::from_str(json_content)
        .context("Failed to parse GitHub event payload as JSON")?;

    Ok(v.pointer("/pull_request/base/sha")
        .and_then(|s| s.as_str())
        .map(|s| s.to_owned()))
}
```